source/extensions/filters/http/dynamic_modules/filter.h

```diff
@@ -166,6 +187,35 @@ class DynamicModuleHttpFilter : public Http::StreamFilter,
 };
 
 using DynamicModuleHttpFilterSharedPtr = std::shared_ptr<DynamicModuleHttpFilter>;
+using DynamicModuleHttpFilterWeakPtr = std::weak_ptr<DynamicModuleHttpFilter>;
+
+/**
+ * This class is used to schedule a HTTP filter event hook from a different thread
+ * than the one it was assigned to. This is created via
+ * envoy_dynamic_module_callback_http_filter_scheduler_new and deleted via
+ * envoy_dynamic_module_callback_http_filter_scheduler_delete.
+ */
+class DynamicModuleHttpFilterScheduler {
+public:
+  DynamicModuleHttpFilterScheduler(DynamicModuleHttpFilterWeakPtr filter,
+                                   Event::Dispatcher& dispatcher)
+      : filter_(std::move(filter)), dispatcher_(dispatcher) {}
+
+  void commit(uint64_t event_id) {
+    dispatcher_.post([filter = filter_, event_id]() {
+      if (DynamicModuleHttpFilterSharedPtr filter_shared = filter.lock()) {
+        filter_shared->onScheduled(event_id);
+      }
+    });
```

Comment:
nit: last comment: I think we shouldn't schedule the event after the onDestroy() is called. There is a little gap between the onDestory() and the destruction of the filter.

But I am OK if you insist on this way.